### PR TITLE
Improve API docs based on Best practices

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/3857

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,8 @@ Ballerina also provides specially designed various database-specific DB connecto
 
 ### Client
 
-The database client should be created using any of the above-listed database libraries and once it is created, the operations and functionality explained below can be used.
+The database client should be created using any of the above-listed database libraries and kept open throughout 
+the entirety of the application to perform the operations and functionality which are explained below.
```
Comment (niveathika) [inline README.md:26]:
```suggestion
A database client can be created using any of the above-listed database libraries. The operations and functionality explained below can be used with the newly created client.

> **Tip** : The client should be used throughout the application lifetime.
```

File: README.md [hunk 2]
```diff
@@ -223,7 +224,8 @@ string|int? generatedKey = result.lastInsertId;
 #### Query data
 
 These samples show how to demonstrate the different usages of the `query` operation to query the
-database table and obtain the results.
+database table and obtain the result as a stream. Once the `query` operation is done, make sure to consume
+all the fetched data or close the stream.
```
Comment (niveathika) [inline README.md]:
```suggestion
database table and obtain the result as a stream. 

> ** Tip**: When processing the stream, make sure to consume all fetched data or close the stream.
```

File: README.md [hunk 3]
```diff
@@ -406,7 +408,7 @@ if resultStr is stream<record{}, sql:Error?> {
 }
 check result.close();
 ```
-Note that you have to invoke the close operation explicitly on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
+Note that you have to invoke the `close` operation explicitly on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
```
Comment (niveathika) [inline README.md:412]:
```suggestion
> **Note**: Once the results are processed, invoke the `close` method on the `sql:ProcedureCallResult` to release the connection resources.
```

File: README.md (hunk 1 above)
Comment (niveathika) [reply README.md:26]: Lets add another note to the Close the client section.

Comment (niveathika) [reply README.md:26]: `Handle connection pools` should also have a tip to configure this as per the need rather than using default values.

File: ballerina/client.bal [hunk 4]
```diff
@@ -21,6 +21,7 @@ import ballerina/jballerina.java;
 public type Client client object {
 
     # Executes the query, which may return multiple results.
+    # Once the operation is done, make sure to consume all the fetched data or close all the streams.
```
Comment (niveathika) [inline ballerina/client.bal:23]:
```suggestion
    # When processing the stream, make sure to consume all fetched data or close the stream.
```

File: ballerina/client.bal [hunk 5]
```diff
@@ -52,7 +53,8 @@ public type Client client object {
     # + return - Metadata of the query execution as an `sql:ExecutionResult[]` or an `sql:Error`
     remote isolated function batchExecute(ParameterizedQuery[] sqlQueries) returns ExecutionResult[]|Error;
 
-    # Executes an SQL query, which calls a stored procedure. This may or may not return results.
+    # Executes an SQL query, which calls a stored procedure. This may or may not
+    # return results as a `sql:ProcedureCallResult` which should be closed after performing this operation.
```
Comment (niveathika) [inline ballerina/client.bal:56]:
```suggestion
    # Executes an SQL query, which calls a stored procedure. This may or may not
    # return results. Once the results are processed, invoke the `close` method on the `sql:ProcedureCallResult` to release the connection resources.
```

Comment (niveathika) [reply ballerina/client.bal:56]: Alos lets add another not on `sql: ProcedureCallResult` close method

File: ballerina/client.bal [hunk 6]
```diff
@@ -16,11 +16,11 @@
 
 import ballerina/jballerina.java;
 
-# Represents an SQL client.
-#
+# Represents an SQL client. It should be kept open throughout the entirety of the application to perform the operations.
```
Comment (niveathika) [inline ballerina/client.bal:19]:
```suggestion
# Represents an SQL client.
```
Lets not add it here, it should be added to the constructor. Lets add it to other database connectors.

File: ballerina/client.bal [hunk 7]
```diff
@@ -52,7 +52,9 @@ public type Client client object {
     # + return - Metadata of the query execution as an `sql:ExecutionResult[]` or an `sql:Error`
     remote isolated function batchExecute(ParameterizedQuery[] sqlQueries) returns ExecutionResult[]|Error;
 
-    # Executes an SQL query, which calls a stored procedure. This may or may not return results.
+    # Executes an SQL query, which calls a stored procedure. This may or may not
+    # return results. Once the results are processed, invoke the `close` method on the `sql:ProcedureCallResult` or
+    # close the current result by calling `getNextQueryResult()` on the `sql:ProcedureCallResult`.
```
Comment (niveathika) [inline ballerina/client.bal:57]:
```suggestion
    # return results. Once the results are processed, invoke the `close` method on the `sql:ProcedureCallResult` to close the connection.
```
`getNextQueryResult()` only closing the current result set and fetches the next. Does not close the connection.

File: README.md [hunk 8]
```diff
@@ -40,6 +42,10 @@ connection pool handling.  For its properties and possible values, see the [`sql
                                new ("jdbc:mysql://localhost:3306/testdb", 
                                 "root", "root");
     ```
+   
+   > **Tip** : Connection pooling is used to optimize opening and closing connections to the database. 
+   > It can add up over time and waste precious network time. Therefore, it is best to open a limited 
+   > connection pool and reuse connections according to the purpose rather than using default.
```
Comment (niveathika) [inline README.md:48]:
```suggestion
> **Tip**: Connection pooling is used to optimize opening and closing connections to the database. However, the pool comes with an overhead. It is best to configure the connection pool properties as per the application need to get the best performance.
```
Move this to line 33. Just after the first paragraph.

File: README.md [hunk 9]
```diff
@@ -85,6 +91,8 @@ Or
 check dbClient.close();
 ```
 
+> **Tip** : The client has to be closed at the end of the application lifetime.
```
Comment (niveathika) [inline README.md:94]:
```suggestion
> **Tip** : The client must be closed only at the end of the application lifetime (or closed for graceful stops in a service).
```

File: ballerina/client.bal (hunk 7 above)
Comment (niveathika) [reply ballerina/client.bal:57]: Add a similar warning here, https://lib.ballerina.io/ballerina/sql/1.6.1/classes/ProcedureCallResult#close

File: README.md [hunk 10]
```diff
@@ -31,6 +31,10 @@ A database client can be created using any of the above-listed database librarie
 All database libraries share the same connection pooling concept and there are three possible scenarios for
 connection pool handling.  For its properties and possible values, see the [`sql:ConnectionPool`](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool).
 
+> **Tip** : Connection pooling is used to optimize opening and closing connections to the database.
+> It can add up over time and waste precious network time. Therefore, it is best to open a limited
+> connection pool and reuse connections according to the purpose rather than using default.
```
Comment (niveathika) [inline README.md:36]: Use a single line. This will format as 3 separate boxes

File: ballerina/client.bal [hunk 11]
```diff
@@ -52,7 +52,8 @@ public type Client client object {
     # + return - Metadata of the query execution as an `sql:ExecutionResult[]` or an `sql:Error`
     remote isolated function batchExecute(ParameterizedQuery[] sqlQueries) returns ExecutionResult[]|Error;
 
-    # Executes an SQL query, which calls a stored procedure. This may or may not return results.
+    # Executes an SQL query, which calls a stored procedure. This may or may not
+    # return results. Once the results are processed, invoke the `close` method on the `sql:ProcedureCallResult`
```
Comment (niveathika) [inline ballerina/client.bal:56]:
```suggestion
    # Executes an SQL query, which calls a stored procedure. This may or may not
    # return results. Once the results are processed, invoke the `close` method on the `sql:ProcedureCallResult`.
```

File: README.md [hunk 12]
```diff
@@ -22,13 +22,17 @@ Ballerina also provides specially designed various database-specific DB connecto
...
+A database client can be created using any of the above-listed database libraries. The operations and functionality explained below can be used with the newly created client.
+
+> **Tip** : The client should be used throughout the application lifetime.
 
 #### Handle connection pools
 
 All database libraries share the same connection pooling concept and there are three possible scenarios for
 connection pool handling.  For its properties and possible values, see the [`sql:ConnectionPool`](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool).
 
+> **Tip** : Connection pooling is used to optimize opening and closing connections to the database. It can add up over time and waste precious network time. Therefore, it is best to open a limited connection pool and reuse connections according to the purpose rather than using default.
```
Comment (niveathika) [inline README.md:34]: Did you revert the changes here?

Comment (niveathika) [reply README.md:34]: https://github.com/ballerina-platform/module-ballerina-sql/pull/604#discussion_r1055535778

Comment (kalaiyarasiganeshalingam) [reply README.md:34]: This was fixed in [7a61380](https://github.com/ballerina-platform/module-ballerina-sql/pull/604/commits/7a61380413be5a99d5ad1577a4cfa114bf568d8b)

Comment (niveathika) [reply README.md:34]: Have you seen the latest changes, seems you have reverted back to the old sentences you used. This suggests connection pool will take up time and waste resources, when in reality it doesnt. Shall we go with the suggested changes, or are you planning on keeping this?

Comment (kalaiyarasiganeshalingam) [reply README.md]: Sorry, I haven't noticed you updated the message. I moved my one.

**Review by niveathika — APPROVED** [review]
